src/js/trove/iolib.js

```diff
@@ -2,46 +2,34 @@
     requires: [],
     provides: {
       values: {
-        "prompt": "tany"
+        "input": "tany"
       }
     },
-    nativeRequires: ["fs"],
-    theModule: function(RUNTIME, NAMESPACE, uri, fs) {
-      // https://github.com/nodejs/node/issues/28243#issuecomment-502402453
-      function PromptIO(msg) {
-        // console.log(RUNTIME);
-        fs.writeSync(RUNTIME.stdout_fd, msg);
-        let s = '';
-        while (true) {
-          let buf = Buffer.alloc(1);
+    nativeRequires: ["readline"],
+    theModule: function(RUNTIME, NAMESPACE, uri, readline) {
+      function Input(msg) {
+          return RUNTIME.pauseStack(function(restarter) {
+            const rl = readline.createInterface({
+              input: process.stdin,
+              output: process.stdout
+            });
 
-          // TODO: figure out why this needs to be within a try-catch
-          try {
-            // TODO: make sure this line works on non-unix OS
-            fs.readSync(RUNTIME.stdin_fd, buf, 0, 1, null);
-          } catch (e) {
-            if (e.code !== "EAGAIN") {
-              throw e;
-            }
-          }
-
-          // TODO: elaborate on more exit conditions
-          if (buf.toString("utf-8") === "\n") {
-            break;
-          }
-
-          s += buf.toString("utf-8");
-        }
-        return s.trim();
+            new Promise(resolve => {
+              rl.question(msg, input => resolve(input));
+            }).then(result => { 
+              restarter.resume(RUNTIME.makeString(result));
+            }).catch(error => {
+              restarter.resume(RUNTIME.makeString(error));
```

Comment:
This feels problematic to me -- it means you're suppressing the error and making both the normal case and the error case to have the same behavior, namely of returning a string.

Reply:
Agreed - silently failing is terrible (found out some code wasn't running in prod once 4 months later because of a silent failure...). Will handle this more correctly.

Reply:
I've updated this, but will revisit after we have consensus on a proper test strategy - see below.